cognite/client/_api/sequences.py

```diff
@@ -0,0 +1,403 @@
+
+        Returns:
+            None
+
+        Examples:
+
+                >>> from cognite.client.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> # with datetime objects
+                >>> data = {123 : ['str',3], 456 : ['bar',42] }
+                >>> res1 = c.sequences.data.insert(columns=['stringColumn','intColumn'], rows=data, id=1)
+        """
+        utils._auxiliary.assert_exactly_one_of_id_or_external_id(id, external_id)
+        base_obj = self._process_ids(id, external_id, wrap_ids=True)[0]
+        base_obj.update(self._process_columns(columns))
+
+        all_rows = [{"rowNumber": k, "values": v} for k, v in rows.items()]
+        row_objs = [
+            {"rows": all_rows[i : i + self._SEQ_POST_LIMIT]} for i in range(0, len(all_rows), self._SEQ_POST_LIMIT)
+        ]
+        tasks = [({**base_obj, **rows},) for rows in row_objs]
+        summary = utils._concurrency.execute_tasks_concurrently(
+            self._insert_data, tasks, max_workers=self._config.max_workers
+        )
+        summary.raise_compound_exception_if_failed_tasks()
+
+    def _insert_data(self, task):
+        self._post(url_path=self._DATA_PATH, json={"items": [task]})
+
+    def delete(self, rows: List[int], id: int = None, external_id: str = None) -> None:
+        """Delete rows from a sequence
+
+        Args:
+            rows: list of row numbers
+            id (int): Id of sequence to delete rows from
+            external_id (str): External id of sequence to delete rows from
+
+        Returns:
+            None
+
+        Examples:
+
+                >>> from cognite.client.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> res1 = c.sequences.data.delete(id=0, rows=[1,2,42])
+        """
+        utils._auxiliary.assert_exactly_one_of_id_or_external_id(id, external_id)
+        post_obj = self._process_ids(id, external_id, wrap_ids=True)[0]
+        post_obj["rows"] = rows
+
+        self._post(url_path=self._DATA_PATH + "/delete", json={"items": [post_obj]})
+
+    def retrieve(
+        self,
+        start: int = 0,
+        end: int = None,
+        columns: List[Union[int, str]] = None,
+        external_id: str = None,
+        id: int = None,
+    ) -> List[dict]:
+        """Retrieve data from a sequence
+
+        Args:
+            start: (inclusive) row number to start from
+            end: (exclusive) upper limit on the row number
+            columns: list of id or external id for the columns of the sequence. Defaults to all if None is passed.
+            id (int): Id of sequence
+            external_id (str): External id of sequence
+
+        Returns:
+            List of sequence data
+        """
+        col, data = self._retrieve_with_columns(start, end, columns, external_id, id)
+        return data
+
+    def retrieve_dataframe(
+        self,
+        start: int = 0,
+        end: int = None,
```

Comment:
start and end should not have default values, makes the code easier to read when used. Also nice for consistency with ts api.

Reply:
API has default limits like this (0 lower, no upper)
Sequences are not time series, and getting the entire (small, static) sequence is likely a common usecase

Reply:
I'm sure it is common usecase. Just as it is common to request all the data in a time series (start=0, end="now"). 
But if i read the two following lines

`retrieve_dataframe(id=1)`
`retrieve_dataframe(id=1, start=0, end=None)`

It is a lot more clear in the second one what is being fetched. Explicit > implicit.

Reply:
or sensible defaults > long copy-pasted argument list, but if you insist.

Reply:
We hade this as defaults for time series datapoints before. Discussions were had - it was agreed that fetching all the data is not always a sensible default. We want to make things intuitive yes, but we also want to avoid abuse of our APIs.